File: sdk/cooldb.py
import requests

class CoolDBClient:
    def __init__(self, base_url="http://localhost:8080"):
        self.base_url = base_url

    def set(self, key, value):
        """Sets a key-value pair in the database."""
        url = f"{self.base_url}/set"
        data = {"key": key, "value": value}
        response = requests.post(url, json=data)

        if response.status_code == 200:
            return response.json().get("message")
        else:
            raise Exception(f"Failed to set key-value pair: {response.json().get('error')}")

    def get(self, key):
        """Gets the value for a given key."""
        url = f"{self.base_url}/get/{key}"
        response = requests.get(url)

        if response.status_code == 200:
            return response.json().get("value")
        elif response.status_code == 404:
            return None
        else:
            raise Exception("Failed to retrieve value for key")

    def list_keys(self):
        """Lists all keys in the database."""
        url = f"{self.base_url}/list"
        response = requests.get(url)

        if response.status_code == 200:
            return response.json().get("keys", [])
        else:
            raise Exception("Failed to list keys")

    def status(self):
        """Checks the status of the database."""
        url = f"{self.base_url}/status"
        response = requests.get(url)

        if response.status_code == 200:
            return response.json().get("status")
        else:
            raise Exception("Failed to retrieve status")

# Example usage
if __name__ == "__main__":
    client = CoolDBClient()

    # Set key-value pair
    print(client.set("hello", "world"))

    # Get value by key
    print(client.get("hello"))

    # List all keys
    print(client.list_keys())

    # Check status
    print(client.status())

File: sdk/cooldb.js
export class CoolDBClient {
	#baseUrl;
	
	constructor(baseUrl = "http://localhost:8080") {
  	this.#baseUrl = baseUrl;
  }
  
  async #fetchJson(endpoint, options = {}) {
    const defaultHeaders = {
      "Accept": "application/json",
    	"Content-Type": "application/json"
    };
    
    try {
    	const response = await fetch(`${this.#baseUrl}${endpoint}`, {
      	...options,
        headers: {
        	...defaultHeaders,
          ...options.headers
        }
      });
      
      const json = await response.json();
      
      if (response.ok) {
      	return json;
      }
      
      throw new Error(json.error ?? `HTTP error - status: ${response.status}`);
    } catch (error) {
    	if (error.message) throw new Error(`API request failed: ${error.message}`);
      
      throw new Error('Network request failed');
    }
  }
  
  async set(key, value) {  	
  	const json = await this.#fetchJson("/set", {
    	method: "POST",
      body: JSON.stringify({ key, value })
    });
    return json.value;
  }
  
  async get(key) {  	
  	if (!key) throw new Error("Key is required");
    const json = await this.#fetchJson(`/get/${encodeURIComponent(key)}`);
    return json.value;
  }
  
  async list() {
  	const json = await this.#fetchJson("/list");
    return json.keys;
  }
  
  async status() {  	
  	const json = await this.#fetchJson("/status");
    return json.status;
  }
}
